### PR TITLE
MM-12791 Fix switching to DM/GMs from notifications and other places

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Using the action by accident instead of the selector. Added some tests to cover this case.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12791

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: tests/components/channel_identifier_router/actions.test.js [hunk 1]
```diff
@@ -53,4 +54,40 @@ describe('Actions', () => {
             expect(emitChannelClickEvent).toHaveBeenCalledWith(channel2);
         });
     });
+
+    describe('goToDirectChannelByUserId', () => {
+        test('switch to a direct channel by user id on the same team', async () => {
+            const testStore = await mockStore(initialState);
```
Comment (sudheerDev) [inline tests/components/channel_identifier_router/actions.test.js]: Do mockStore need to be async?

File: utils/utils.jsx [hunk 2]
```diff
@@ -1368,10 +1369,10 @@ export function getUserIdFromChannelName(channel) {
 }
 
 // Used to get the id of the other user from a DM channel id (id1_id2)
-export function getUserIdFromChannelId(channelId) {
+export function getUserIdFromChannelId(channelId, currentUserId = getCurrentUserId(store.getState())) {
```
Comment (sudheerDev) [inline utils/utils.jsx]: Nit: Do we need this as second argument for this func?

File: tests/components/channel_identifier_router/actions.test.js (hunk 1 above)
Comment (jwilander) [reply tests/components/channel_identifier_router/actions.test.js]: I'm not entirely sure, I was just copying the usages I saw elsewhere

File: utils/utils.jsx (hunk 2 above)
Comment (jwilander) [reply utils/utils.jsx]: Yes, we shouldn't really be accessing the store directly in any utils functions but I didn't want to have to fix every usage of this function to pass in the current user ID, hence the default parameter

**Pull request merged**